### ISSUE TITLE
sphinx-autogen broken with 2.3.0

### ISSUE THREAD

**Issue by dopplershift** [sphinx-doc/sphinx#6936]
**Describe the bug**
I'm completely unable to run the `sphinx-autogen` script. When I do, I get:
```pytb
Traceback (most recent call last):
  File "/Users/rmay/miniconda3/envs/py38/lib/python3.8/site-packages/sphinx/ext/autosummary/generate.py", line 142, in skip_member
    return app.emit_firstresult('autodoc-skip-member', objtype, name,
AttributeError: 'DummyApplication' object has no attribute 'emit_firstresult'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/rmay/miniconda3/envs/py38/bin/sphinx-autogen", line 10, in <module>
    sys.exit(main())
  File "/Users/rmay/miniconda3/envs/py38/lib/python3.8/site-packages/sphinx/ext/autosummary/generate.py", line 465, in main
    generate_autosummary_docs(args.source_file, args.output_dir,
  File "/Users/rmay/miniconda3/envs/py38/lib/python3.8/site-packages/sphinx/ext/autosummary/generate.py", line 280, in generate_autosummary_docs
    rendered = generate_autosummary_content(name, obj, parent,
  File "/Users/rmay/miniconda3/envs/py38/lib/python3.8/site-packages/sphinx/ext/autosummary/generate.py", line 182, in generate_autosummary_content
    get_members(obj, {'function'}, imported=imported_members)
  File "/Users/rmay/miniconda3/envs/py38/lib/python3.8/site-packages/sphinx/ext/autosummary/generate.py", line 163, in get_members
    skipped = skip_member(value, name, documenter.objtype)
  File "/Users/rmay/miniconda3/envs/py38/lib/python3.8/site-packages/sphinx/ext/autosummary/generate.py", line 145, in skip_member
    logger.warning(__('autosummary: failed to determine %r to be documented.'
  File "/Users/rmay/miniconda3/envs/py38/lib/python3.8/logging/__init__.py", line 1797, in warning
    self.log(WARNING, msg, *args, **kwargs)
  File "/Users/rmay/miniconda3/envs/py38/lib/python3.8/site-packages/sphinx/util/logging.py", line 124, in log
    super().log(level, msg, *args, **kwargs)
  File "/Users/rmay/miniconda3/envs/py38/lib/python3.8/logging/__init__.py", line 1829, in log
    self.logger.log(level, msg, *args, **kwargs)
  File "/Users/rmay/miniconda3/envs/py38/lib/python3.8/logging/__init__.py", line 1500, in log
    self._log(level, msg, args, **kwargs)
  File "/Users/rmay/miniconda3/envs/py38/lib/python3.8/logging/__init__.py", line 1577, in _log
    self.handle(record)
  File "/Users/rmay/miniconda3/envs/py38/lib/python3.8/logging/__init__.py", line 1587, in handle
    self.callHandlers(record)
  File "/Users/rmay/miniconda3/envs/py38/lib/python3.8/logging/__init__.py", line 1649, in callHandlers
    hdlr.handle(record)
  File "/Users/rmay/miniconda3/envs/py38/lib/python3.8/logging/__init__.py", line 946, in handle
    rv = self.filter(record)
  File "/Users/rmay/miniconda3/envs/py38/lib/python3.8/logging/__init__.py", line 807, in filter
    result = f.filter(record)
  File "/Users/rmay/miniconda3/envs/py38/lib/python3.8/site-packages/sphinx/util/logging.py", line 381, in filter
    self.app._warncount += 1
AttributeError: 'DummyApplication' object has no attribute '_warncount'
```

The part about `emit_firstresult` and `autodoc-skip-member` make me think #6817 is related. In general, the problem seems to be that the command-line script relies on `DummyApplication`, which is missing some attributes, and maybe functionality.

I can say that if I add `autosummary_generate=True` and `autosummary_imported_members=True` to `conf.py` and just run everything through `sphinx-build` then it all works as before, which is really just fine with me. It seems like the `sphinx-autogen` script is just borked.

**To Reproduce**
```
$ git clone https://github.com/Unidata/MetPy
$ cd MetPy
$ pip install .
$ cd docs
$ sphinx-autogen api/index.rst
```

**Expected behavior**
The autosummary generation script to work like it has in previous versions.

**Environment info**
- OS: Linux and Mac
- Python version: 3.8.0
- Sphinx version: 2.3.0
- Sphinx extensions:  sphinx.ext.autodoc, sphinx.ext.autosummary

**Comment by tk0miya** [comment]
Thank you for reporting. Indeed, this is broken... I'll fix this ASAP. Please wait a moment.

**Comment by tk0miya** [comment]
Fixed by #6941 .
Thank you for reporting. I'll publish a bugfix package later (maybe within a day)!
